src/sql/src/plan/decorrelate.rs

```diff
@@ -427,6 +454,32 @@ impl RelationExpr {
             }
         }
     }
+
+    /// Computes whether this RelationExpr is safe to apply directly in a a
+    /// branch.
+    fn is_branch_safe(&self) -> bool {
+        let mut is_safe = true;
+        // Exhaustive match to prevent accidents when expression variants
+        // change.
+        self.visit(&mut |expr| match expr {
+            RelationExpr::Distinct { .. }
+            | RelationExpr::Reduce { .. }
+            | RelationExpr::Threshold { .. }
+            | RelationExpr::Negate { .. }
```

Comment:
I wasn't sure that these two need to be in here. Do you have a sense for if they need to be, or if they just weren't obviously safe?

Reply:
The latter! Anyway, reworked, see incoming comment.